src/overlays/render_settings.rs

```diff
@@ -343,6 +350,39 @@ impl Overlay for RenderSettingsOverlay {
 
                 maps.current_map = current_map;
 
+                // Move the Globals from the old scene to the new scene
+                if map_changed {
+                    // We have learned the power to Take worlds
+                    let mut old_scene = take(&mut maps.map_mut(old_map_index).unwrap().scene);
+
+                    let mut ent_list = vec![];
+
+                    for (entity, global) in old_scene.query::<Option<&Global>>().iter() {
+                        if let Some(g) = global {
+                            if g.0 {
+                                ent_list.push(entity);
+                            }
+                        }
+                    }
+
+                    let mut selected = resources.get_mut::<SelectedEntity>().unwrap();
+                    if let Some(map) = maps.current_map_mut() {
+                        for entity in ent_list {
+                            let new_ent = map.scene.reserve_entity();
+                            map.scene
+                                .insert(new_ent, old_scene.take(entity).ok().unwrap())
+                                .ok();
+                            if selected.0.is_some_and(|e| e == entity) {
+                                selected.0.replace(new_ent);
+                            }
```

Comment:
Can be simplified, `Option<T>` is comparable with others of the same type
```suggestion
                            if selected.0 == Some(entity) {
                                selected.0.replace(new_ent);
                            }
```